app/assets/javascripts/scribing_canvas.js

```diff
@@ -0,0 +1,284 @@
+    $('#scribing-delete-' + qid)
+      .click({ canvas: c }, function (event) {
+        var canvas = event.data.canvas;
+
+        if(canvas.getActiveGroup()) {
+          canvas.getActiveGroup().forEachObject(function(o){ canvas.remove(o) });
+          canvas.discardActiveGroup().renderAll();
+        }
+        else {
+          canvas.remove(canvas.getActiveObject());
+        }
+      });
+
+    $('#scribing-zoom-in-' + qid)
+      .click({ canvas: c }, function (event) {
+          var canvas = event.data.canvas;
+
+          var newZoom = canvas.getZoom() + 0.1;
+          canvas.zoomToPoint({ x: canvas.height/2, y: canvas.width/2 }, newZoom);
+      });
+
+    $('#scribing-zoom-out-' + qid)
+      .click({ canvas: c }, function(event) {
+        var canvas = event.data.canvas;
+
+        var newZoom = Math.max(canvas.getZoom() - 0.1, 1);
+        canvas.zoomToPoint({ x: canvas.height/2, y: canvas.width/2 }, newZoom);
+      });
+
+    $('#scribing-layers-' + qid)
+      .change(function () {
+          $(this).find('option').each(function (i, o) {
+            var show_layer = $(o).attr('selected') == 'selected';
+            $(o).data('toggle_layer')(show_layer);
+          });
+      }); 
+
+  });
+
+  //assign each canvas its image
+  $.each(scribingImages, function(qid, scribingImage) {
+    //get appropriate canvas by qid
+    var c = fabricCanvases[qid];
+
+    //calculate scaleX and scaleY to fit image into canvas before creating fabric.Image object
+    var scale = Math.min(c.width / scribingImage.width, c.height / scribingImage.height, 1);
+
+    //create fabric.Image object with the right scaling and set as canvas background
+    var fabricImage = new fabric.Image(scribingImage, {opacity: 1, scaleX: scale, scaleY: scale});
+    c.setBackgroundImage(fabricImage, c.renderAll.bind(c));
+    c.renderAll();
+
+    var layersList = $('#scribing-layers-' + qid);
+    layersList.selectpicker('hide'); // remains hidden if there are no layers
+
+    var load_scribble = function (scribble) {
+      // from http://jsfiddle.net/Kienz/sFGGV/6/ via https://github.com/kangax/fabric.js/issues/704
+      if (scribble.val() == '') {
+        return;
+      }
+ 
+      // Convert javascript objects to fabricjs objects
+      var objects = JSON.parse(scribble.val()).objects;
+      var drawn_items = [];
+      for (var i = 0; i < objects.length; i++) {
+        var klass = fabric.util.getKlass(objects[i].type);
+        if (klass.async) {
+          klass.fromObject(objects[i], function (img) {
+            drawn_items.push(img);
+          });
+        } else {
+          var item = klass.fromObject(objects[i]);
+          drawn_items.push(item);
+        }
+      }      
+      
+      // Case when scribble should be read-only    
+      if (scribble.data('locked') && drawn_items.length > 0) {
+        var scribble_group = new fabric.Group(drawn_items);
```

Comment:
Identifier 'scribble_group' is not in camel case.<br>Identifier 'drawn_items' is not in camel case.